travel-application-development-guide.mdx

```diff
@@ -0,0 +1,1594 @@
+            </Link>
+          ))}
+          …
+        </div>
+      </div>
+    </>
+  );
+}
+```
+
+For this code block, import the Xata client and the component handling, deleting both the trip and note detail at the same time where the id is passed as a property, `<BtnDelTrip id={data.id} note={notes[0]?.id} />` to identify the ID of the linked table trip ID and the note ID.
+
+In addition, we filter the records of the notes column from the database using only the ID with the `.filter()` method, `const notes = await xata.db.notes.filter("trip.id", data?.id).getMany()`. Loop through the notes with the `.map()` method and access the columns of each Note record using their properties attached to the Link component to open the note on a different route.
+
+![Trip and note detail](/images/travel-app-guide/trip-note-detail.jpeg)
+
+Create these files `components/BtnDelTrip.js` and `utils/delete-trip.js`. Copy-paste the file content with the code in the [BtnDelTrip](https://github.com/Terieyenike/track-trip-dashboard-with-xata-next/blob/main/components/BtnDelTrip.js) and [delete-trip](https://github.com/Terieyenike/track-trip-dashboard-with-xata-next/blob/main/utils/delete-trip.js) files, respectively.
+
+![Delete trip detail](/images/travel-app-guide/delete-trip-detail.jpeg)
+
+## Deploy to Vercel
+
+Vercel can access a GitHub repository with these steps and deploy the project.
+
+- Push all the application code to GitHub locally
+- On your Vercel dashboard, create a New Project
+- Link the new project with the GitHub repository using the Import button
+- In Settings, update the Environment Variables to match with your local `.env` file
+- Deploy
+
+## Conclusion
+
+This guide walked through the seamless integration of Xata and Next.js to build a travel-tracking application. With these technologies, you can now relive and document all the key moments from your trips. Integrating Tailwind and React Toastify helped to create a better user interface and experience.
+
+The union between Xata and Next.js helps developers build dynamic and serverless applications, whether you are an experienced developer or a novice. It is an all-encompassing way to deliver a 360º approach to full-stack CRUD applications.
+
+Happy coding!
+
+More information
+For more detailed insights, explore the references in this post below:
+
+- [Demo track trips (https://track-trip-dashboard-with-xata-next.vercel.app/)]
+- [GitHub repo (https://github.com/Terieyenike/track-trip-dashboard-with-xata-next)]
+- [Xata file attachments (https://xata.io/docs/sdk/file-attachments)]
+- [Route groups in Next.js (https://nextjs.org/docs/app/building-your-application/routing/route-groups)]
+- [Next.js with Xata (https://xata.io/docs/getting-started/nextjs)]
+- [Add notifications to apps (https://github.com/fkhadra/react-toastify#readme)]
+- [Fetching, caching, revalidating (https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating)]
```

Comment:
```suggestion
- [Fetching, caching, revalidating](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating)
```

Reply:
Thanks, @CezzaineZ. I had to revert the file to the initial commit where the code snippets were correct. That's why some things went back to what @alexfrancoeur already fixed. I'm currently trying to fix some errors with code snippets. But once I finish I'll fix the rest.